src/zenml/cli/secret.py

```diff
@@ -1118,3 +1132,51 @@ def delete_secret(name_or_id: str, yes: bool = False) -> None:
             )
         except NotImplementedError as e:
             error(f"Centralized secrets management is disabled: {str(e)}")
+
+
+@secret.command("export", help="Export a secret as a YAML file.")
+@click.argument(
+    "name_id_or_prefix",
+    type=click.STRING,
+)
+@click.option(
+    "--scope",
+    "-s",
+    type=click.Choice([scope.value for scope in list(SecretScope)]),
+    default=None,
+)
+@click.option(
+    "--filename",
+    "-f",
+    type=click.STRING,
+    default=None,
+    help=(
+        "The name of the file to export the secret to. Defaults to "
+        "<secret_name>.yaml."
+    ),
+)
+def export_secret(
+    name_id_or_prefix: str,
+    scope: Optional[str] = None,
+    filename: Optional[str] = None,
+) -> None:
+    """Export a secret as a YAML file.
+
+    The resulting YAML file can then be imported as a new secret using the
+    `zenml secret create <new_secret_name> -v @<filename>` command.
+
+    Args:
+        name_id_or_prefix: The name of the secret to export.
+        scope: The scope of the secret to export.
+        filename: The name of the file to export the secret to.
+    """
+    from zenml.utils.yaml_utils import write_yaml
+
+    secret = _get_secret(name_id_or_prefix=name_id_or_prefix, scope=scope)
+    if not secret.secret_values:
+        warning(f"Secret with name `{name_id_or_prefix}` is empty.")
+        return
```

Comment:
Is this how we want to do it? If someone makes some script to backup their secrets, and we only create secret.yaml files if there are values contained inside, then potentially they lose those secrets (at least the knowledge that they had set one up etc). Would it be better practice to just output a file with an empty secret value instead?

Reply:
agreed